README.md

```diff
@@ -206,6 +207,39 @@ class SimpleMetricsLibrary: MetricsFactory {
 }
 ```
 
+### Managing metric lifecycle
+
+Since the Metrics API needs to accomodate various types of library implementations, care has been given to enable
+libraries which require explicitly managing the lifecycle of their metrics receive signals about a metric no longer being 
+"in use" by other libraries.
+
+Not all libraries will have a need for this lifecycle management, e.g. if `MetricHandler` implementations are stateless
+forwarders to some backend system, it is likely that managing lifecycle of individual metrics objects may not be required.
+However, in case of libraries which keep any created `MetricHandler` in some internal registry (e.g. in the form of a 
+dictionary `[MetricId: MetricHandler]`), it may be necessary to invoke `MetricsSystem.release()` for known not-used-anymore 
+metrics in order to facilitate eagerly discarding this state from the registry.
+
+The eager releasing of metrics plays an even more important role for `MetricsHandlers` which rely on "heavy" resources
+such as: file handles, network connections or large in-memory data structures (e.g. high definition resolution histograms,
+which can take tens, up to hundreds of KiB).
```

Comment:
For reference https://github.com/HdrHistogram/HdrHistogram#footprint-estimation and that's a fairly efficient one (without losing precision)